joomla/joomla-cms
# Allow using underscore character in permission name

Pull Request for Issue #9645 .
#### Summary of Changes

This PR fix an issue causes by some changes in saving permissions in Joomla 3.5.0. In older version of Joomla, we are allowed to use underscore character in permission name. From 3.5.0, if we use underscore character, the permission won't be saved properly. Please see https://github.com/joomla/joomla-cms/issues/9645 to understand the issue
#### Testing Instructions
1. Open the file administrator/components/com_content/access.xml
2. Add a new permission which has underscore character (for example content.view_content) to component section. Example: `<action name="content.view_content" title="View Content" description="Allows viewing articles" />`
3. Go to System -> Global Configuration, select Articles component, Permissions tab (http://localhost/joomla35/administrator/index.php?option=com_config&view=component&component=com_content).
4. Try to change permission for View Content permission
5. Before this patch, you could not change setting for the new permission you just added
6. After this patch, the change you made for the  new permission can be saved
7. Try to change other permission and make sure it is saved properly

Hopefully, this issue can be fixed and go to 3.5.1
